client/db/src/lib.rs

```diff
@@ -1483,7 +1490,12 @@ impl<Block: BlockT> sc_client_api::backend::Backend<Block> for Backend<Block> {
 		})
 	}
 
-	fn revert(&self, n: NumberFor<Block>, revert_finalized: bool) -> ClientResult<NumberFor<Block>> {
+	fn revert(
```

Comment:
This function should additionally check that the state we are reverting to is still available (by calling  `self.have_state_at()`)